EIPS/eip-draft-whisper.md

```diff
@@ -0,0 +1,176 @@
+## Preamble
+
+    EIP: draft
+    Title: Whisper Specification
+    Author: Vlad Gluhovsky <gluk256@gmail.com>
+    Type: Informational
+    Status: Draft
+    Created: 2017-05-05
+
+## Abstract
+
+This draft EIP describes the format of Whisper messages within the ÐΞVp2p Wire Protocol.
+This EIP should substitute the [existing specification](https://github.com/ethereum/wiki/wiki/Whisper-Wire-Protocol).
+More detailed documentation on Whisper could be found [here](https://github.com/ethereum/go-ethereum/wiki/Whisper).
+
+## Motivation
+
+It is necessary to specify the standard for Whipser messages in order to ensure forward compatibility of different Whisper clients.
+
+## Specification
+
+All Whisper messages sent as ÐΞVp2p Wire Protocol packets should be RLP-encoded arrays of data containing two objects: integer packet code followed by another object (whose type depends on the packet code). 
+
+If Whisper node does not support a particular packet code, it should just ignore the packet without generating any error.
+
+### Packet Codes
+
+The message codes reserved for Whisper protocol: 0 - 127.
+Messages with unknown codes must be ignored, for forward compatibility of future versions.
+
+The Whisper sub-protocol should support the following packet codes:
+
+| EIP   | Name                       | Int Value |
+|-------|----------------------------|-----------|
+|       | Status                     |     0     |
+|       | Messages                   |     1     |
+|       | PoW Requirement            |     2     |
+|       | Bloom Filter               |     3     |
+|-------|----------------------------|-----------|
+
+
+The following message codes are optional, but they are reserved for specific purpose.
+
+| EIP   | Name                       | Int Value |
+|-------|----------------------------|-----------|
+|       | P2P Request                |    126    |
+|       | P2P Message                |    127    |
+|-------|----------------------------|-----------|
+
+### Packet Format and Usage
+
+**Status** [`0`]
+
+This packet contains two objects: integer (0x00). Might be followed by some arbitrary data in future versions, which should be gracefully ignored for future compatibility.
```

Comment:
The first sentence says this packet contains two objects.  The second sentence reads like the second object might not exist.  Does this packet always contains two objects, or sometimes just one object?